docs/openapi.yaml

```diff
@@ -85,6 +85,57 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/Problem"
+    put:
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NviPeriod'
```

Comment:
Consider using the request body evaluators:
x-amazon-apigateway-request-validators

Reply:
This gives you the perk of faster bad request responses. -> bad request response ~ 5ms
And less need for validating the request body inside the lambda. -> faster 201 response
And this ensures that your openapi spec is not just theoretical, it's part of the actual implementation.

Reply:
aws request validation documentation: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-request-validation-sample-api-swagger.html

Reply:
Have added body required, but it seems that there is no string validation, only number value can be validated.